### PR TITLE
dev: sync checker with x/tools@v0.28.0

### PR DESCRIPTION
**Pull request by ldez**
Updates the code borrowed from x/tools

Continuation of the work done inside #5109

I still have not found the Lost Ark but I am near the Temple of Doom :smile_cat:

### CODE REVIEW

File: pkg/goanalysis/runner_checker.go [hunk 1]
```diff
@@ -94,106 +104,140 @@ func (act *action) analyze() {
...
-		analyzedIn := time.Since(startedAt)
-		if analyzedIn > time.Millisecond*10 {
+		analyzedIn := time.Since(t1)
+		if analyzedIn > 10*time.Millisecond {
 			debugf("%s: run analyzer in %s", act, analyzedIn)
 		}
-	}
 
-	// disallow calls after Run
+		// correct result type?
```
Comment (bombsimon) [inline pkg/goanalysis/runner_checker.go:192]: What is the question here? Why they're different? The code seems to properly check the difference and return a well format error. Should the comment indicate that it's not known why they would be different?

**Review by bombsimon — APPROVED** [review]
LGTM! 🚢

File: pkg/goanalysis/runner_checker.go (hunk 1 above)
Comment (ldez) [reply pkg/goanalysis/runner_checker.go:192]: it's a copy of the original comment :wink: 

https://github.com/golang/tools/blob/v0.28.0/go/analysis/checker/checker.go#L360

Comment (bombsimon) [reply pkg/goanalysis/runner_checker.go:192]: 🤦 Oh sorry. I thought it was you since it was a lone diff, should've checked upstream first!

**Pull request merged**